code/game/objects/items/grenades/grenade.dm

```diff
@@ -44,6 +44,9 @@
 	sleep(det_time)//so you dont die instantly
 	return BRUTELOSS
 
+/obj/item/grenade/ex_act(severity, target) // Little boom can chain a big boom.
+	detonate()
```

Comment:
Call `..()`

Reply:
Or hell, even just register to `COMSIG_ATOM_EX_ACT` with `detonate` yourself if you want after making `detonate` a SIGNAL_HANDLER.